quinn-proto/src/connection/assembler.rs

```diff
@@ -218,6 +355,30 @@ impl PartialEq for Buffer {
     }
 }
 
+#[derive(Debug, Eq, PartialEq, Ord, PartialOrd)]
+enum BufferState {
+    Fragmented(NonZeroUsize),
+    Defragmented,
+}
+
+impl BufferState {
+    fn new(size: usize, allocation_size: usize, possibly_duplicate: bool) -> Self {
+        if possibly_duplicate || Self::significant_over_allocation(size, allocation_size) {
+            match NonZeroUsize::new(allocation_size) {
+                Some(value) => Self::Fragmented(value),
+                None => Self::Defragmented,
```

Comment:
Can that ever be true? Would it be worthwhile to have an `unreachable!` here, and to make sure adding an empty buffer is a no-op?

Reply:
This got also removed in my latest changes, sorry for that.